### ISSUE TITLE
Socks5 proxy

### ISSUE THREAD

**Issue by Ahmad-2213** [wiresock/WireSockUI#64]
The Connection can not be stablished via Socks5 proxy.
![image](https://github.com/wiresock/WireSockUI/assets/106820058/845a7be8-acc9-4b2c-bfd0-61792a1a133e)

**Comment by wiresock** [comment]
Could you please try the build 1.2.39 linked to [this](https://www.ntkernel.com/forums/topic/new-wiresockui-install-socks5-issue/page/2/) forum topic

**Comment by Ahmad-2213** [comment]
> Could you please try the WireSock VPN Client build 1.2.39 linked to [this](https://www.ntkernel.com/forums/topic/new-wiresockui-install-socks5-issue/page/2/) forum topic?

Can't Open the wiresock-client.exe on Windows 10 .
Should I Try to uninstall the previous versions and then try again?

**Comment by wiresock** [comment]
I don't think I understand what is happening. Have you installed the [this build](https://www.wiresock.net/downloads/wiresock-vpn-client-x64-1.2.39.1.msi)? If anything goes wrong try uninstalling the software, reboot and install it again.

**Comment by Ahmad-2213** [comment]
> I don't think I understand what is happening. Have you installed the [this build](https://www.wiresock.net/downloads/wiresock-vpn-client-x64-1.2.39.1.msi)? If anything goes wrong try uninstalling the software, reboot and install it again.

I tried all of it , But still Can't run the program...
![image](https://github.com/wiresock/WireSockUI/assets/106820058/a0d28fbb-7340-4af2-a0b8-ba79c56ffa35)

**Comment by wiresock** [comment]
The application in the screenshot is the command line WireSock VPN Client, which lacks a graphical user interface. For more information, visit [this link](https://www.wiresock.net/wiresock-vpn-client/wiresock-vpn-client-cli/). If you prefer a graphical interface, please also download and install WireSockUI which will run as an UI front end.

**Comment by Ahmad-2213** [comment]
> The application in the screenshot is the command line WireSock VPN Client, which lacks a graphical user interface. For more information, visit [this link](https://www.wiresock.net/wiresock-vpn-client/wiresock-vpn-client-cli/). If you prefer a graphical interface, please also download and install WireSockUI which will run as an UI front end.

There's no actual UI for this version.
Do you have any suggestions?

**Comment by wiresock** [comment]
1. Download and install WireSock VPN Client v1.2.39 using the link above.
2. Download and run the latest available WireSockUI.

**Comment by Ahmad-2213** [comment]
> 1. Download and install WireSock VPN Client v1.2.39 using the link above.
> 2. Download and run the latest available WireSockUI.

Still doesn't work , I guess I'm doing something wrong.
Don't mean to bother you anymore...

**Comment by wiresock** [comment]
Try removing all components, restarting your system, and starting the installation process from scratch. The only difference now is that you are using a more recent build of the WireSock VPN CLI, which was not an issue when you initially got it running successfully.

**Comment by Ahmad-2213** [comment]
> 1. Download and install WireSock VPN Client v1.2.39 using the link above.
> 2. Download and run the latest available WireSockUI.

So I removed All of the components and did the same as you said, however, It's still not working...
![image](https://github.com/wiresock/WireSockUI/assets/106820058/75fc8d9c-9c6d-4db2-8281-d4c87616e35d)

**Comment by Ahmad-2213** [comment]
and this is captured with CLI version:
![image](https://github.com/wiresock/WireSockUI/assets/106820058/81d2b6e0-da40-4ef1-a8ad-952688f23ecb)

**Comment by wiresock** [comment]
Please note that uou should not use square brackets in the command line. If path does not contain spaces you can also omit the quotes.

**Comment by Ahmad-2213** [comment]
> Please note that uou should not use square brackets in the command line. If path does not contain spaces you can also omit the quotes.

I fixed that. Still encountering troubles with connecting to the config.
I'll also put my config down here to see if everything's right!
![image](https://github.com/wiresock/WireSockUI/assets/106820058/288851a9-75f6-4ee8-a6a4-70373578de62)
![image](https://github.com/wiresock/WireSockUI/assets/106820058/65472f13-7ded-4f93-9360-622e8465c309)

**Comment by wiresock** [comment]
Ah, I see the issue now. You are attempting to use a SOCKS5 proxy at the address 127.0.0.1:2080. Please note that the localhost address is not supported for this purpose. If this is a standard SOCKS5 proxy, using it in this manner is generally ineffective as it will simply relay the original handshake. However, if the SOCKS5 proxy is configured with Shadowsocks, this setup might be useful. I will consider adding support for such configurations in the future.

**Comment by Ahmad-2213** [comment]
> Ah, I see the issue now. You are attempting to use a SOCKS5 proxy at the address 127.0.0.1:2080. Please note that the localhost address is not supported for this purpose. If this is a standard SOCKS5 proxy, using it in this manner is generally ineffective as it will simply relay the original handshake. However, if the SOCKS5 proxy is configured with Shadowsocks, this setup might be useful. I will consider adding support for such configurations in the future.

I really appreciate that. my last question is :
is this going to tunnel the handshake or it'll detour all of the data through the proxy?
similarly in OpenVPN , the proxy is used to work as a chain meaning that the traffic is passed into the proxy and then going to the server.
my point is that I want to bypass the censorship of TLS Handshake by tunneling via proxy , not to route all of my data through it.

**Comment by wiresock** [comment]
Only the Wireguard handshake is routed through the SOCKS5 proxy because it is the most easily detected part of the protocol. All other Wireguard data packets are transmitted directly. You can refer my [post](https://www.ntkernel.com/how-to-bypass-egypts-wireguard-ban/) for the details.

**Comment by Ahmad-2213** [comment]
> Ah, I see the issue now. You are attempting to use a SOCKS5 proxy at the address 127.0.0.1:2080. Please note that the localhost address is not supported for this purpose. If this is a standard SOCKS5 proxy, using it in this manner is generally ineffective as it will simply relay the original handshake. However, if the SOCKS5 proxy is configured with Shadowsocks, this setup might be useful. I will consider adding support for such configurations in the future.

yes I'm using Shadowsocks on Nekoray Client.

**Comment by Ahmad-2213** [comment]
![image](https://github.com/wiresock/WireSockUI/assets/106820058/45eeff14-d70a-4e0b-843a-145d29c3a49f)

**Comment by Ahmad-2213** [comment]
Hello There! 
are there any scheduals for new update?

**Comment by wiresock** [comment]
I don't have an estimate for local SOCKS5 proxy support yet, but it is on my to-do list.

**Comment by MHMoosaei** [comment]
> Hello There! are there any scheduals for new update?

I am also looking forward to the addition of local SOCKS5 proxy support feature, but until then, there is a workaround that somewhat addresses this issue. In the WireSock settings, add the Nekoray app and its used cores in the Disallowed apps section so that Nekoray does not use this WireSock connection. Then, set Nekoray to Tun mode and activate the desired proxy. Now you can activate WireSock.

DisallowedApps = nekobox_core,nekoray,xray,sing-box

**Comment by Ahmad-2213** [comment]
> > Hello There! are there any scheduals for new update?
> 
> I am also looking forward to the addition of local SOCKS5 proxy support feature, but until then, there is a workaround that somewhat addresses this issue. In the WireSock settings, add the Nekoray app and its used cores in the Disallowed apps section so that Nekoray does not use this WireSock connection. Then, set Nekoray to Tun mode and activate the desired proxy. Now you can activate WireSock.
> 
> DisallowedApps = nekobox_core,nekoray,xray,sing-box

Well , the benefits of using SOCKS5 proxy is that We don't need to Tunnel the traffic through Nekoray's Core. Instead , we just tunnel the handshake to go through proxy to keep the connection alive.
The case you're mentioning just slows down the Connection. better way is to just pass the handshake through Nekoray , Not all of the Data.

**Comment by MHMoosaei** [comment]
> > > Hello There! are there any scheduals for new update?
> > 
> > 
> > I am also looking forward to the addition of local SOCKS5 proxy support feature, but until then, there is a workaround that somewhat addresses this issue. In the WireSock settings, add the Nekoray app and its used cores in the Disallowed apps section so that Nekoray does not use this WireSock connection. Then, set Nekoray to Tun mode and activate the desired proxy. Now you can activate WireSock.
> > DisallowedApps = nekobox_core,nekoray,xray,sing-box
> 
> Well , the benefits of using SOCKS5 proxy is that We don't need to Tunnel the traffic through Nekoray's Core. Instead , we just tunnel the handshake to go through proxy to keep the connection alive. The case you're mentioning just slows down the Connection. better way is to just pass the handshake through Nekoray , Not all of the Data.

Yes, that's correct. The method I mentioned is only temporary until the local SOCKS5 proxy support feature is added to the application. Additionally, you can use a proxy that another device on your local network has shared.

**Comment by wiresock** [comment]
Current profile validation and the bundled template support the WireSock SOCKS5 proxy directives, credentials, and `Socks5ProxyAllTraffic`. Closing this older report as addressed.
